### PR TITLE
[preflight-check] Checking the possibility of opening ports 6443 2379 2380 on the server (#5070)

### PR DESCRIPTION
**Pull request by aleksey-su**
## Description
Checking the possibility of opening ports 6443 2379 2380 on the server and connection to them

## Why do we need it, and what problem does it solve?
issue #5070 

## What is the expected result?
Preventing failures before bootstrap execution

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: dhctl
type: feature
summary: Checking the possibility of opening ports 6443 2379 2380 on the server and connection to them before bootstrap 
impact_level: default
```

### CODE REVIEW

**Comment by aleksey-su** [comment]
image
image

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've started the workflow for label 'e2e/run/openstack'...

File: candi/bashible/preflight/check_ports.sh [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    return $exit_code
+}
+
+for port in 6443 2379 2380
+do
+    echo -n "Check port $port "
+    check_port $port
+    if [ $? -ne 0 ]; then
+        echo "FAIL"
+        exit 1
```
Comment (name212) [inline candi/bashible/preflight/check_ports.sh:45]: please accumulate errors

File: dhctl/pkg/preflight/ports.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	"strings"
+
+	"github.com/deckhouse/deckhouse/dhctl/pkg/app"
+	"github.com/deckhouse/deckhouse/dhctl/pkg/log"
+	"github.com/deckhouse/deckhouse/dhctl/pkg/template"
+)
+
+func (pc *preflightCheck) CheckAvailabilityPorts() error {
+	if app.PreflightSkipAvailabilityPorts {
+		log.InfoLn("Skip availability ports preflight check")
```
Comment (name212) [inline dhctl/pkg/preflight/ports.go:29]: availability ports preflight check
```suggestion
		log.InfoLn("Availability ports preflight check was skipped")
```

**Review by name212 — CHANGES_REQUESTED** [review]
some comments

**Comment by aleksey-su** [comment]
image
image

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've started the workflow for label 'e2e/run/static'...

**Comment by aleksey-su** [comment]
/e2e/abort static;Static;containerd;1.23 5574230140 5574230140-2-con-1-23 /sys/deckhouse-oss/install:pr5195 astra@192.168.199.231

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by aleksey-su** [comment]
image
image

**Comment by z9r5** [comment]
/e2e/abort static;Static;containerd;1.23 5574230140 5574230140-1-con-1-23 /sys/deckhouse-oss/install:pr5195 astra@192.168.199.114

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by z9r5** [comment]
/e2e/abort static;Static;containerd;1.23 5574230140 5574230140-1-con-1-23 /sys/deckhouse-oss/install:pr5195 astra@192.168.199.114

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've started the workflow for label 'e2e/run/static'...

**Review by RomanenkoDenys — APPROVED** [review]
LGTM

**Review by name212 — APPROVED** [review]

**Pull request merged**